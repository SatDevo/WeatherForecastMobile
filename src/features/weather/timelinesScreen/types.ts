interface WeatherInterval {
  startTime: string;
  values: {
    precipitationIntensity: number;
    precipitationType: number;
    windSpeed: number;
    windGust: number;
    windDirection: number;
    temperature: number;
    temperatureApparent: number;
    cloudCover: number;
    cloudBase: number | null;
    cloudCeiling: number | null;
    weatherCode: number;
  };
}

export interface WeatherData {
  data: {
    timelines: Array<{
      timestep: string;
      startTime: string;
      endTime: string;
      intervals: WeatherInterval[];
    }>;
  };
}

export interface ProcessedWeatherData {
  city: string;
  country: string;
  current: {
    temperature: string;
    condition: string;
    windSpeed: string;
    humidity: string;
    sunrise: string;
  };
  forecast: Array<{
    time: string;
    temp: string;
    condition: 'sun' | 'moderaterain' | 'heavyrain' | 'partlycloudy' | 'cloud' | 'mist';
    windSpeed: string;
  }>;
}