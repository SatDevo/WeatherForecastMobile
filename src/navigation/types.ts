import { SCREEN_ALIASES } from '../constants/screenAliases';

export type RootStackParamList = {
  [SCREEN_ALIASES.TIMELINES]: undefined;
};

export type TimelinesStackParamList = {
  [SCREEN_ALIASES.TIMELINES_ENTRY]: undefined;
};

