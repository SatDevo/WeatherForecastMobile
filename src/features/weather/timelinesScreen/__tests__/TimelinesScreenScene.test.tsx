import React from 'react';
import renderer from 'react-test-renderer';
import { TimelinesScreenScene } from '../TimelinesScreen.scene';
import { ProcessedWeatherData } from '../types';

// Mock the required components
jest.mock('../../../../components/atoms/IconButton', () => ({
  IconButton: 'IconButton',
}));
jest.mock('../../../../components/molecules/LocationDisplay', () => ({
  LocationDisplay: 'LocationDisplay',
}));
jest.mock('../../../../components/molecules/TemperatureDisplay', () => ({
  TemperatureDisplay: 'TemperatureDisplay',
}));
jest.mock('../../../../components/molecules/WeatherMetric', () => ({
  WeatherMetric: 'WeatherMetric',
}));
jest.mock('../../../../components/organisms/DailyForecastList', () => ({
  DailyForecastList: 'DailyForecastList',
}));

// Mock react-native-feather icons
jest.mock('react-native-feather', () => ({
  Search: 'Search',
  Wind: 'Wind',
  Droplet: 'Droplet',
  Sun: 'Sun',
}));

// Mock the COLORS constant
jest.mock('../../../../constants/colors', () => ({
  COLORS: {
    white: '#FFFFFF',
  },
}));

describe('TimelinesScreenScene', () => {
  const mockWeatherData: ProcessedWeatherData = {
    city: 'New York',
    country: 'USA',
    current: {
      temperature: 25,
      condition: 'Sunny',
      windSpeed: '10',
      humidity: '60',
      sunrise: '06:00',
    },
    forecast: [
      { time: '12:00', temp: '26°', condition: 'Sunny', windSpeed: '12 km/h' },
      { time: '15:00', temp: '28°', condition: 'Partly Cloudy', windSpeed: '14 km/h' },
    ],
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <TimelinesScreenScene weatherData={mockWeatherData} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays correct location', () => {
    const instance = renderer.create(
      <TimelinesScreenScene weatherData={mockWeatherData} />
    ).root;

    const locationDisplay = instance.findByType('LocationDisplay');
    expect(locationDisplay.props.city).toBe('New York');
    expect(locationDisplay.props.country).toBe('USA');
  });

  it('displays correct current weather information', () => {
    const instance = renderer.create(
      <TimelinesScreenScene weatherData={mockWeatherData} />
    ).root;

    const temperatureDisplay = instance.findByType('TemperatureDisplay');
    expect(temperatureDisplay.props.temperature).toBe('25°');
    expect(temperatureDisplay.props.condition).toBe('Sunny');

    const weatherMetrics = instance.findAllByType('WeatherMetric');
    expect(weatherMetrics[0].props.value).toBe('10 km/h');
    expect(weatherMetrics[1].props.value).toBe('60%');
    expect(weatherMetrics[2].props.value).toBe('06:00');
  });

  it('passes forecast data to DailyForecastList', () => {
    const instance = renderer.create(
      <TimelinesScreenScene weatherData={mockWeatherData} />
    ).root;

    const dailyForecastList = instance.findByType('DailyForecastList');
    expect(dailyForecastList.props.forecasts).toEqual(mockWeatherData.forecast);
  });
});

