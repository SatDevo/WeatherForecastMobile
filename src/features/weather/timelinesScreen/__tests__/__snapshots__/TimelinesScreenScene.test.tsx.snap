// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelinesScreenScene renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 60,
        "paddingBottom": 60,
        "paddingHorizontal": 20,
      }
    }
  >
    <IconButton
      onPress={[Function]}
    >
      <Search
        height={24}
        stroke="#FFFFFF"
        width={24}
      />
    </IconButton>
  </View>
  <LocationDisplay
    city="New York"
    country="USA"
  />
  <TemperatureDisplay
    condition="Sunny"
    iconCondition="sun"
    temperature="25°"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": 40,
        "paddingHorizontal": 20,
      }
    }
  >
    <WeatherMetric
      icon={
        <Wind
          height={20}
          stroke="#FFFFFF"
          width={20}
        />
      }
      value="10 km/h"
    />
    <WeatherMetric
      icon={
        <Droplet
          height={20}
          stroke="#FFFFFF"
          width={20}
        />
      }
      value="60%"
    />
    <WeatherMetric
      icon={
        <Sun
          height={20}
          stroke="#FFFFFF"
          width={20}
        />
      }
      value="06:00"
    />
  </View>
  <DailyForecastList
    forecasts={
      [
        {
          "condition": "Sunny",
          "temp": "26°",
          "time": "12:00",
          "windSpeed": "12 km/h",
        },
        {
          "condition": "Partly Cloudy",
          "temp": "28°",
          "time": "15:00",
          "windSpeed": "14 km/h",
        },
      ]
    }
  />
</View>
`;
